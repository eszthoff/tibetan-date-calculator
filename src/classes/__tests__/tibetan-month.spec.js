import MockDate from 'mockdate';
import TibetanMonth from '../tibetan-month';

describe('TibetanMonth', () => {
  beforeAll(() => {
    MockDate.set('2019/06/07');
  });
  afterAll(() => {
    MockDate.reset();
  });

  it('should create a correct class when invoked without arg', () => {
    expect(new TibetanMonth()).toMatchSnapshot();
  });
  it('should create a correct class when invoked with string arg', () => {
    expect(new TibetanMonth('2019/06/22')).toMatchSnapshot();
  });
  it('should create a correct class when invoked with object arg', () => {
    expect(new TibetanMonth({
      year: 2146, month: 3, isLeapMonth: false
    })).toMatchSnapshot();
  });
  it('should not calculate days when initially incoved', () => {
    const month = new TibetanMonth();
    expect(month.days.length).toEqual(0);
  });
  it('should not calculate days when requested (includes skipped and doubled days)', () => {
    const month = new TibetanMonth({ year: 2146, month: 1 });
    const days = month.getDays();

    expect(month.days.length).not.toEqual(0);
    expect(days.length).toMatchSnapshot();
  });
  it('should return year object correctly)', () => {
    const month = new TibetanMonth();
    const { yearObj } = month;

    expect(yearObj).toMatchSnapshot();
  });
});
