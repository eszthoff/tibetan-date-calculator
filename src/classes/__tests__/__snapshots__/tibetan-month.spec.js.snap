// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TibetanMonth should create a correct class when invoked with object arg 1`] = `
TibetanMonth {
  "days": Array [],
  "endDateStr": "2019-06-03",
  "isDoubledMonth": false,
  "isLeapMonth": false,
  "month": 3,
  "startDateStr": "2019-05-05",
  "year": 2146,
}
`;

exports[`TibetanMonth should create a correct class when invoked with string arg 1`] = `
TibetanMonth {
  "days": Array [],
  "endDateStr": "2019-07-02",
  "isDoubledMonth": false,
  "isLeapMonth": false,
  "month": 4,
  "startDateStr": "2019-06-04",
  "year": 2146,
}
`;

exports[`TibetanMonth should create a correct class when invoked without arg 1`] = `
TibetanMonth {
  "days": Array [],
  "endDateStr": "2019-07-02",
  "isDoubledMonth": false,
  "isLeapMonth": false,
  "month": 4,
  "startDateStr": "2019-06-04",
  "year": 2146,
}
`;
